### PR TITLE
Add support for ESP32 OTA

### LINKED ISSUE #934: Is the Update Feature working ?
I had TasmoAdmin for over 2 years and currently on version 3.07 on docker (with compose).

I installed it to help update my Tasmota devices. for over 2 years it has never worked for me. Not once.

I'm giving this a last try before deleting it.

When I try to update, TasmoAdmin says update is successful. I tried the automatic feature where it fetches the latest Tasmota version itself and I tried manual and upload the minimal and full versions. Both never worked.

Below are the screenshots.

image

image

image

On the device console I see the below lines:
```
stat/tasmota_3C9CC2/RESULT = {"Upgrade":"Version 12.4.0 from http://172.18.0.2:80/data/firmwares/tasmota.bin"}
stat/tasmota_3C9CC2/UPGRADE = {"Upgrade":"Failed HTTP error: connection failed"}
```

While TasmoAdmin says the update was successful, the device fails to update. I tried on all devices, same story.

TasmoAdmin sounds useful but it never worked for me so I would appreciate any help.

Thank you.

### PR DESCRIPTION
**Pull request by inverse**
Add support for ESP32 based firmwares.

image

Resolves #934

### CODE REVIEW

**Comment by inverse** [comment]
@BlueWings172  how does the OTA update process work for ESP32? for ESP8266 it requires minimal but looks like ESP32 doesnt use that and doesnt support gz binaries

File: tasmoadmin/src/Helper/TasmotaHelper.php [hunk 1]
```diff
@@ -86,8 +88,34 @@ public function getReleases(): array
 
     public function getLatestFirmwares(string $configuredFirmware): AutoFirmwareResult
     {
-        $firmwareResult = $this->getLatestRelease();
+        $isEsp32 = str_starts_with($configuredFirmware, 'tasmota32');
```
Comment (inverse) [inline tasmoadmin/src/Helper/TasmotaHelper.php:91]: This assumes that if the firmware starts with this it's esp32 - not sure it's the best thing to assume, but seems to work :/

**Comment by inverse** [comment]
@alexdelprete do you have any experience with esp32 tasmota devices?

**Comment by BlueWings172** [comment]
@inverse The Sonoff Zigbee Bridge Pro is the only ESP32 I have flashed with Tasmota and I did not use minimal version or gz files. That said, flashing this particular device for the first time is not a straightforward process and requires expanding the FS partition and uploading several files. So I wish I had more answers.

I followed mostly this guide. https://notenoughtech.com/home-automation/tasmota-on-sonoff-zb-bridge-pro/#flash

**Comment by alexdelprete** [comment]
> @alexdelprete do you have any experience with esp32 tasmota devices?

Yes I use them for the Nuki Hub and for the Blueconnect measurements of the pool water.

There's no minimal fw, and no gzipped version, AFAIK. ESP32 is not constrained like esp82xx devices.

I use mainly M5Stack Atom devices.

**Comment by alexdelprete** [comment]
> @inverse The Sonoff Zigbee Bridge Pro is the only ESP32 I have flashed with Tasmota and I did not use minimal version or gz files. That said, flashing this particular device for the first time is not a straightforward process and requires expanding the FS partition and uploading several files. So I wish I had more answers.
> 
> I followed mostly this guide. https://notenoughtech.com/home-automation/tasmota-on-sonoff-zb-bridge-pro/#flash

I flash them all the time, the most simple method is the tasmota web installer: https://tasmota.github.io/install/

It always worked for me. As an alternative, when I compile a custom tasmota version, I also use espressif's ESPtool from CLI: https://github.com/espressif/esptool

example: esptool.exe --port com3 --baud 1500000 write_flash 0x0 tasmota32.bin

There's no requirement to use the three separate fw files (boot, partition, etc.).

**Comment by inverse** [comment]
Thanks both of you for the input! So it sounds like what's provided will fulfil the needs for just the OTA update part which is the main intention of this extension.

**Comment by alexdelprete** [comment]
> So it sounds like what's provided will fulfil the needs for just the OTA update part which is the main intention of this extension.

Right now I'm not able to test it, the 3 ESP32 devices all use a custom compiled tasmota firmware, so I will manage it with TasmoAdmin, but I won't upgrade the fw with it, I need to keep it custom. BTW: would be nice to have a do not update flag for a device, so when you do batch updates, you're confident those are not touched. :)

**Comment by inverse** [comment]
@alexdelprete created https://github.com/TasmoAdmin/TasmoAdmin/issues/940

**Pull request merged**